doc/source/serve/deployment-graph/control_flow_based_on_user_inputs.md

```diff
@@ -13,6 +13,24 @@ The example shows how to user inputs to control the graph flow
    :language: python
 ```
 
+```{note}
+1. The dag.execute() take arbitrary number of arguments, and internally the graph will access inputs by index (e.g. user_input[0], user_input[1]).
+
+2. value1 and value2 are ObjectRef passed into the combine, the value of ObjectRef will be addressed at the runtime.
+
+3. we can pass value1 and value2 as a list. In this case, we are passing the ObjectRef as reference, the value of ObjectRef will not be addressed automatically. We need to explicitly use ray.get() to address value before we do sum() or max() function.
```

Comment:
maybe link to our object ref docs here for pass by reference ? https://docs.ray.io/en/latest/ray-core/objects.html?#passing-objects-by-reference